diffkemp/simpll/CPatternPass.h

```diff
@@ -0,0 +1,55 @@
+//===------ CPatternPass.h - preprocessing pass for custom C patterns -----===//
+//
+//       SimpLL - Program simplifier for analysis of semantic difference      //
+//
+// This file is published under Apache 2.0 license. See LICENSE for details.
+// Author: Tomas Kucma, tomaskucma2@gmail.com
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the CPatternPass class, which provides
+/// preprocessing for custom C patterns.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef DIFFKEMP_SIMPLL_CPATTERNPASS_H
+#define DIFFKEMP_SIMPLL_CPATTERNPASS_H
+
+#include "Utils.h"
+#include <llvm/IR/Module.h>
+#include <tuple>
+#include <unordered_map>
+
+using namespace llvm;
+
+/// Class responsible for preprocessing custom C patterns, which are already
+/// compiled to LLVM IR. Primary function consists of renaming functions to
+/// proper names and tagging pattern starts and ends.
+class CPatternPass {
```

Comment:
Originally I planned to run the pass purely from the ```opt``` tool, however, since the pass is utilizing the SimpLL library itself, there was some weird behavior because both SimpLL library and opt dynamically linked to LLVM 16 library, and incorrect versions of functions were called. Even now, when it's called from the SimpLL code, I think there's an issue with the fact that the comparison function called from the pass also runs passes and I think running a pass from the inside of another pass was problematic.

Reply:
Right, I see. Then, the question is what happens if we just use `DifferentialFunctionComparator` for the analysis. I'm still not sure if we need to run all passes that the entire analysis runs.